flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionFactory.java

```diff
@@ -227,23 +263,78 @@ public ResultPartition create(
             }
         } else if (type == ResultPartitionType.HYBRID_FULL
                 || type == ResultPartitionType.HYBRID_SELECTIVE) {
-            partition =
-                    new HsResultPartition(
-                            taskNameWithSubtaskAndId,
-                            partitionIndex,
-                            id,
-                            type,
-                            subpartitions.length,
-                            maxParallelism,
-                            batchShuffleReadBufferPool,
-                            batchShuffleReadIOExecutor,
-                            partitionManager,
-                            channelManager.createChannel().getPath(),
-                            networkBufferSize,
-                            getHybridShuffleConfiguration(numberOfSubpartitions, type),
-                            bufferCompressor,
-                            isBroadcast,
-                            bufferPoolFactory);
+            if (tieredStorageConfiguration != null) {
+                // Create memory manager.
+                TieredStorageMemoryManager memoryManager =
+                        new TieredStorageMemoryManagerImpl(
+                                checkNotNull(tieredStorageConfiguration)
+                                        .getNumBuffersTriggerFlushRatio(),
+                                true);
+
+                // Create buffer accumulator.
+                int accumulatorExclusiveBufferNum =
```

Comment:
Maybe use `numAccumulatorExclusiveBuffers` to replace `accumulatorExclusiveBufferNum`.